src/pytopmod/core/dcel/mesh.py

```diff
@@ -0,0 +1,77 @@
+import dataclasses
+from typing import Generator
+
+from pytopmod.core import keystore
+from pytopmod.core import mesh as base_mesh
+from pytopmod.core.edge import EdgeKey
+from pytopmod.core.face import FaceKey
+from pytopmod.core.vertex import VertexKey
+
+
+@dataclasses.dataclass(slots=True)
+class EdgeNode:
+    """An Edge Node in the DCEL structure."""
+
+    vertex_1_key: VertexKey
+    vertex_2_key: VertexKey
+    face_1_key: FaceKey
+    face_2_key: FaceKey
+    vertex_1_next_key: EdgeKey
+    vertex_2_next_key: EdgeKey
+
+    def __repr__(self) -> str:
+        return dataclasses.astuple(self).__repr__()
+
+
+@dataclasses.dataclass(slots=True)
+class Mesh(base_mesh.Mesh):
+    """DCEL-backed Mesh class.
+
+    This structure uses a map of EdgeNodes.
+
+    Manifold-preserving operators are implemented in 'operators.py'.
+    """
+
+    edge_keys: keystore.KeyStore[EdgeKey] = dataclasses.field(init=False)
+    edge_nodes: dict[EdgeKey, EdgeNode] = dataclasses.field(init=False)
+
+    def __post_init__(self):
+        super(Mesh, self).__post_init__()
+        self.edge_keys = keystore.KeyStore[EdgeKey]("e")
+        self.edge_nodes = {}
+
+    def create_edge(
+        self,
+        vertex_1_key: VertexKey,
+        vertex_2_key: VertexKey,
+        face_1_key: FaceKey,
+        face_2_key: FaceKey,
+        vertex_1_next_key: EdgeKey,
```

Comment:
vertex_{1|2}_next_key are edges, not a vertices

Reply:
Yes they correspond to the "next edge at vertex_1|2", do you think we should rename the member to something like `vertex_1_next_edge_key` ?

Reply:
I dont think we need to complicate, edge_1 _key and edge_2_key should be enough. It shows that we expect edge keys there, and the function description can explain what these are.

Reply:
Agreed. Changing to `edge_1_key` and `edge_2_key` for this PR. I am also considering reverting the change that added the `_key` prefixes as I feel that it reduces readability, plus they feel redundant with type annotations, let me know what you think and I'll prepare another PR to do that.

Reply:
In the last commit, changed to `next_edge_{1|2}_key`